test_rclcpp/test/test_intra_process.cpp

```diff
@@ -67,7 +67,7 @@ TEST(test_intra_process_within_one_node, nominal_usage) {
 
     // wait for the first callback
     printf("spin_node_once() - callback (1) expected\n");
-    executor.spin_node_once(node);
+    executor.spin_node_some(node);
```

Comment:
I think this is a race condition now. Before `spin_node_once` was blocking until the callback was triggered. Now with `spin_node_some` the callback might not yet be ready.